crates/sargon/src/radix_connect/wallet_interaction/dapp_wallet_interaction/wallet_to_dapp/success_response/pre_authorization/subintent_response.rs

```diff
@@ -52,9 +90,15 @@ mod tests {
     }
 
     #[test]
-    fn new() {
-        let signed_subintent = SignedSubintent::sample();
-        let sut = SUT::new(signed_subintent);
-        assert_eq!(sut, SUT::sample());
+    fn json_roundtrip() {
+        assert_eq_after_json_roundtrip(
+            &SUT::sample_other(),
+            r#"
+            {
+                "signedPartialTransaction": "4d220e03210221012105210607f20a00000000000000000a0a000000000000002200002200000ab168de3a00000000202000220000202000202200202100202200202000",
+                "subintentHash": "subtxid_sim1kdwxe9mkpgn2n5zplvh4kcu0d69k5qcz679xhxfa8ulcjtjqsvtq799xkn"
+            }
+            "#,
+        );
     }
 }
```

Comment:
Lets try to see why you did not get derive serde to work. but if we really cannot then:

please add unit tests for FAILING json decoding e.g. missing key `"subintentHash"` and another missing `"signedPartialTransaction"` and some where those keys exist but wrong value kind. That is really what one ought to do when writing a custom serde.

Reply:
keep this simple single `json_roundtrip` test, but put more unit tests of failing serde inside `SignedSubintent` (see https://github.com/radixdlt/sargon/pull/279/files#r1860603707 )

Reply:
actually we don't _need_ the `subintentHash` field to deserialize, but will add tests for the case without `signedPartialTransaction` or where its content is invalid